src/vs/code/node/cli.ts

```diff
@@ -319,23 +319,89 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = args['status'] || args['verbose'] || hasReadStdinArg;
+			const envVars: string[] = [];
+			for (const envKey in env) {
+				if (envKey === '_') {
+					continue;
 				}
-			});
+				const value = env[envKey];
+				envVars.push('--env');
+				envVars.push(envKey + '=' + value ?? '');
+			}
+			const openArgs: string[] = ['-n'];
```

Comment:
This is interesting, the help doc reads:

> Open a new instance of the application even if one is already running

But would this maybe allow us to have a fast way of communicating to a first running instance if there is one? I would assume that the `open` invocation ends up as a `open-file` event on Electron's `app` maybe?

Reply:
If you're referencing the `requestSingleInstanceLock` issue, I think it is easier to go with the current planned approach, which is adding a `data` parameter to the `app.requestSingleInstanceLock()` method. That method also comes with this event https://www.electronjs.org/docs/api/app#event-second-instance

Reply:
@bpasero that is an interesting idea, yup the event will be emitted for `open -n` but right now it only provides the file path that was requested to be opened with new instance as an argument to the event. But if we can expand the event in electron then this would be a cool way to do single instance.

Reply:
Yeah that is what I thought, currently we are missing a lot of context from the second launch in that event.

Reply:
Would expanding on that event only help out macOS? https://www.electronjs.org/docs/api/app#event-open-file-macos

Reply:
Not entirely because we can improve two way communication in this modal, I would suggest to invest on improving `requestSingleInstanceLock`